### ISSUE TITLE
Support `git add -p` on Windows

### ISSUE THREAD

**Issue by lzybkr** [dandavison/delta#40]
If you try setting interactive.diffFilter to delta, you'll get an error:

```
fatal: mismatched output from interactive.diffFilter
hint: Your filter must maintain a one-to-one correspondence
hint: between its input and output lines.
```

If I add `--file-style plain --hunk-style plain` then `git add -p` "works", but there is no coloring at all.

**Comment by dandavison** [comment]
This works now that we have the `--color-only` argument:

```
[interactive]
    diffFilter = delta --color-only
```

<table><tr><td><img width=450px src="https://user-images.githubusercontent.com/52205/80408297-f975b680-8894-11ea-888d-1c869163ef97.png" alt="image" /></td></tr></table>

**Comment by lzybkr** [comment]
It doesn't quite work for me on Windows it seems like the file extension is not detected correctly:

![image](https://user-images.githubusercontent.com/2148248/80409841-69c50d80-887e-11ea-810e-09496e6bcd94.png)

And that same chunk viewed with `git diff`:

![image](https://user-images.githubusercontent.com/2148248/80409810-5ade5b00-887e-11ea-990b-1248c094ef45.png)

**Comment by dandavison** [comment]
@lzybkr are you able to get hold of the raw escape codes that are being emitted there?

>  it seems like the file extension is not detected correctly

My initial guess would be that the pager / terminal emulator combo that is coming into play there is failing to render ASCI escape sequences for background colors correctly, but you're the expert for this on Windows.

A file with no extension gets the usual background colors:

<table><tr><td><img width=300px src="https://user-images.githubusercontent.com/52205/80626044-f0fbb800-8a1b-11ea-8da4-2900ae063d2c.png" alt="image" /></td></tr></table>

**Comment by dandavison** [comment]
@lzybkr is `git add -p` working for you on Windows? I haven't heard any other reports. The above looked like a terminal emulator / less issue to me? (Are you able to get hold of the raw escape codes?)

**Comment by lzybkr** [comment]
I hadn't tried `git add -p` for some time, but trying it now, it seems to work. It's hard to say for sure what changed - I know I've updated `git` (but `less` isn't running so it shouldn't be involved), and I've updated the terminal, but that shouldn't have mattered since I was using the same terminal from WSL and it worked fine.

At any rate, since I can't repro anymore and you don't have other reports, I can close the issue.
